app/src/main/java/org/openobservatory/ooniprobe/data/TestData.java

```diff
@@ -313,6 +347,113 @@ public static void setAnomaly_ndt(String entry, NetworkMeasurement test){
         }
     }
 
+    /*
+     whatsapp: red if "whatsapp_endpoints_status" or "whatsapp_web_status" or "registration_server" are "blocked"
+     docs: https://github.com/TheTorProject/ooni-spec/blob/master/test-specs/ts-018-whatsapp.md#semantics
+     */
+    public static void setAnomaly_whatsapp(String entry, NetworkMeasurement test) {
+        if(!test.entry) {
+            TestStorage.setEntry(context, test);
+            test.entry = true;
+        }
+        try {
+            int anomaly = OONITests.ANOMALY_GREEN;
+            JSONObject jsonObj = new JSONObject(entry);
+            String keys[] = {"whatsapp_endpoints_status",
+                    "whatsapp_web_status",
+                    "registration_server_status"};
+            for (String key: keys)
+            {
+                String value = jsonObj.getString(key);
+                if (value == null)
+                    anomaly = OONITests.ANOMALY_ORANGE;
+                else if (value.equals("blocked"))
+                    anomaly = OONITests.ANOMALY_RED;
+            }
+            if (test.anomaly < anomaly) {
+                test.anomaly = anomaly;
+                TestStorage.setAnomaly(context, test.test_id, anomaly);
+            }
+        } catch (JSONException e) {
+            System.out.println("JSONException "+ e);
```

Comment:
Why are you using `System.out.println` here and `Log.v` above?

I would suggest using always the same functions for logging as it makes debugging logs easier.

Reply:
basically when it's Log.v it was Antonio's code, system.out is mine. I replaced them all.

Reply:
Is there a reason to prefer `System.out` to `Log.v`?